### PR TITLE
Add getMatchesByRequest

### PR DESCRIPTION
**Pull request by mateow99**
## Overview

Added getMatchedByRequest route for when we add the matching algorithm

## Changes Made

- Add getMatchedByRequestId route
- Add "created" field to posts
- Add test case to make sure the route works

## Test Coverage

Created test case and tested route

## Next Steps (delete if not applicable)

opens #47

### CODE REVIEW

File: src/models/PostModel.ts [hunk 1]
```diff
@@ -25,6 +25,9 @@ export class PostModel {
   @Column("text", { array: true })
   images: string[];
 
+  @Column("numeric")
+  created: number;
```
Comment (snajima) [inline src/models/PostModel.ts:29]: is there a reason why we're using `number` type instead of `timestamptz`? https://www.postgresql.org/docs/current/datatype-datetime.html

File: src/services/RequestService.ts [hunk 2]
```diff
@@ -59,4 +60,19 @@ export class RequestService {
       return await requestRepository.deleteRequest(request);
     });
   }
+
+  public async getMatchesByRequestId(params: UuidParam, getMatchesRequest: GetMatchesRequest): Promise<PostModel[]> {
+    return this.transactions.readOnly(async (transactionalEntityManager) => {
+      const requestRepository = Repositories.request(transactionalEntityManager);
+      let request;
+      if (getMatchesRequest.time == 0) {
```
Comment (snajima) [inline src/services/RequestService.ts:68]: when would `time` be 0?

Comment (snajima) [reply src/services/RequestService.ts:68]: i feel like i vaguely remember this being a design choice, but maybe add some documentation somewhere explaining when `time` can be 0 for future resell devs to not be confused

File: src/tests/RequestTest.test.ts [hunk 3]
```diff
@@ -14,6 +15,7 @@ let requestController: RequestController;
 
 beforeAll(async () => {
   await DatabaseConnection.connect();
+  jest.setTimeout(5000);
```
Comment (snajima) [inline src/tests/RequestTest.test.ts:18]: why is this necessary?

File: src/types/ApiRequests.ts [hunk 4]
```diff
@@ -88,7 +89,11 @@ export interface UploadImageRequest {
 // REQUEST
 
 export interface CreateRequestRequest {
-    title: string
+    title: string;
```
Comment (snajima) [inline src/types/ApiRequests.ts:92]: so awkward

**Review by snajima — CHANGES_REQUESTED** [review]
Again, good stuff but curious about your design choice with the `time` field

File: src/services/RequestService.ts (hunk 2 above)
Comment (chalo2000) [reply src/services/RequestService.ts:68]: If you are trying to get from this endpoint and you want to get all requests regardless of time, just make frontend make a request without passing a query parameter. That is clearer than setting the query parameter to 0 which has already caused confusion.

**Review by chalo2000 — APPROVED** [review]
Approving to give @snajima the approval power; I agree with his comments and Archit's that you should be changing time from UNIX to timestamp. I was wondering what the difference is and found that it's to avoid timezone issues (which for AppDev's case would likely be a non-issue in the forseeable future) https://stackoverflow.com/questions/178704/are-unix-timestamps-the-best-way-to-store-timestamps. One helpful case for us would be to have a human readable time format so if we're debugging we don't always need to be using an online UNIX to timestamp converter.

File: src/models/PostModel.ts [hunk 5]
```diff
@@ -25,8 +25,8 @@ export class PostModel {
   @Column("text", { array: true })
   images: string[];
 
-  @Column("numeric")
-  created: number;
+  @CreateDateColumn({ type: 'timestamptz' })
```
Comment (snajima) [inline src/models/PostModel.ts:28]: o my this decorator is crazy good find mateo :+1:

File: src/api/controllers/RequestController.ts [hunk 6]
```diff
@@ -38,8 +38,8 @@ export class RequestController {
     return { request: await this.requestService.deleteRequestById(params) };
   }
 
-  @Post('matches/id/:id/')
-  async getMatchesByRequestId(@Body() getMatchesRequest: GetMatchesRequest, @Params() params: UuidParam): Promise<GetPostsResponse> {
-    return { posts: await this.requestService.getMatchesByRequestId(params, getMatchesRequest) };
+  @Get('matches/id/:id/:time?/')
+  async getMatchesByRequestId(@Params() params: TimeParam): Promise<GetPostsResponse> {
```
Comment (snajima) [inline src/api/controllers/RequestController.ts:42]: correct me if i'm wrong, but does this mean we're inputting the date through the url?

**Review by snajima — CHANGES_REQUESTED** [review]
i guess i'm only slightly confused about how time is being inputted into the `getMatchesByRequestId` endpoint. have you postman tested? if so i'll approve :)

File: src/api/controllers/RequestController.ts (hunk 6 above)
Comment (mateow99) [reply src/api/controllers/RequestController.ts:42]: Yeah we are

**Review by snajima — APPROVED** [review]
ok good stuff! lgtm :)
